plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -804,14 +856,20 @@ function OPDSBrowser:showDownloads(item)
             end
         end
     end
-    if (#type_buttons % 2 == 1) then -- we need even number of type buttons
-        table.insert(type_buttons, {text = ""})
+    -- handles adding download button(s)
+    if (download_buttons ~= {}) then
+        if (#download_buttons == 1) then
+            table.insert(buttons, download_buttons)
+        elseif (#download_buttons % 2 == 1) then
+            table.insert(download_buttons, {text = ""})
+            table.insert(buttons, download_buttons)
```

Comment:
Since you are inserting unconditionally, move it out of the "if".

Reply:
Done.